### ISSUE TITLE
Using scale and rotate together sometimes ends up in an invisible component.

### ISSUE THREAD

**Issue by BrandonMA** [nandorojo/moti#85]
### What's happening?
Whenever you try to apply scale and rotate together the component sometimes renders and performs the animation, and sometimes it doesn't.

This is literally random as sometimes it works and sometimes it doesn't, I am not sure on how the library is implemented but might be some kind of race condition? I also include a video to show the problem plus the code to reproduce, it's literally me just pressing refresh many times.

https://user-images.githubusercontent.com/6232522/124679375-e04d9a00-de89-11eb-86d0-a8c7916345a5.mp4

### Code

```
import React from 'react';
import { View } from 'react-native';
import { MotiView } from 'moti';
import { registerRootComponent } from 'expo';

export function App(): JSX.Element {
    return (
        <View>
            <MotiView
                style={{ backgroundColor: 'blue', height: 300, width: 300 }}
                from={{ opacity: 0, scale: 0, rotate: '0deg' }}
                animate={{ opacity: 1, scale: 1, rotate: '45deg' }}
            />
        </View>
    );
}

export default registerRootComponent(App);
```

If you have an idea on why this is going on but don't have the time to fix it a brief explanation would be nice and I can make a pull request.

### Snack

https://snack.expo.io/@brandonma/moti-rotate-scale-bug

**Comment by nandorojo** [comment]
Hmm very odd. What if you use a `0deg` string instead of numbers for rotate?

It seems like #76 and #83 could be related, but I'm not totally sure why.

**Comment by nandorojo** [comment]
Random idea, does moving the props out of render make a difference? (Doubtful since there are no re-renders in your example)

What if you remove opacity? Do they still not work?

**Comment by nandorojo** [comment]
Does setting `transition={{ type: 'timing' }}` do anything? Maybe it's due to a mix of springs (the transforms) and timing (opacity), though I doubt it.

**Comment by nandorojo** [comment]
Could you put your reproduction on expo snack?

**Comment by BrandonMA** [comment]
> Does setting `transition={{ type: 'timing' }}` do anything? Maybe it's due to a mix of springs (the transforms) and timing (opacity), though I doubt it.

Setting timing for the transition type ends up in no animation in this particular scenario, if you remove scale it performs the animation😅

**Comment by BrandonMA** [comment]
> Hmm very odd. What if you use a `0deg` string instead of numbers for rotate?
> 
> It seems like #76 and #83 could be related, but I'm not totally sure why.

Same effect if I use a string, no animation when using both properties, I will update the example and add it because it should have it.

**Comment by BrandonMA** [comment]
> Random idea, does moving the props out of render make a difference? (Doubtful since there are no re-renders in your example)
> 
> What if you remove opacity? Do they still not work?

Same😞 as you said there's no re renders but even if I take it out it keeps happening.

And opacity doesn't seem to be related, removing it did nothing, but removing scale or rotate actually fixes the issue.

**Comment by BrandonMA** [comment]
> Could you put your reproduction on expo snack?

Yes for sure! I will add the snack to the issue once I finish it.

**Comment by nandorojo** [comment]
Hey just wanted to see if you were able to reproduce this in a snack. I'll close it for now, but more than happy to open it once it's reproduced! Thanks again.

**Comment by BrandonMA** [comment]
> Hey just wanted to see if you were able to reproduce this in a snack. I'll close it for now, but more than happy to open it once it's reproduced! Thanks again.

Actually it's already on the description! But here it is too: https://snack.expo.io/@brandonma/moti-rotate-scale-bug

**Comment by nandorojo** [comment]
Ah thanks I missed that

**Comment by nandorojo** [comment]
Okay so this is interesting. It seems to work fine on Web, but I do see the issue on iOS sometimes...🧐

Here's a video of it working on Web: https://www.loom.com/share/7994f42ad2e54042ba73765506f7b88b

Snack: https://snack.expo.dev/@beatgig/moti-rotate-scale-bug

```tsx
import * as React from 'react';
import { Text, View, Pressable } from 'react-native';
import { MotiView } from 'moti';

export default function App() {
  const [visible, toggle] = React.useReducer((visible) => !visible, true);
  return (
    <Pressable style={{ flex: 1 }} onPress={toggle}>
      {visible && (
        <MotiView
          style={{ backgroundColor: 'blue', height: 300, width: 300 }}
          from={{ scale: 0, rotate: '0deg' }}
          animate={{ scale: 1, rotate: '45deg' }}
        />
      )}
    </Pressable>
  );
}
```

**Comment by nandorojo** [comment]
# Temporary Solution

Use `useAnimationState` instead of the `from` & `animate` props.

## Example

This works: 

```tsx
import * as React from 'react';
import { Text, View, Pressable } from 'react-native';
import { MotiView, useAnimationState } from 'moti';

const from = {
  scale: 0,
  rotate: '0deg',
};

const animate = {
  scale: 1,
  rotate: '45deg',
};

export default function App() {
  const state = useAnimationState({
    from,
    to: animate,
  });

  return (
      <MotiView
        style={{ backgroundColor: 'blue', height: 300, width: 300 }}
        state={state}
      />
  );
}
```

Yet this does _not_ work:

```tsx
import * as React from 'react';
import { Text, View, Pressable } from 'react-native';
import { MotiView, useAnimationState } from 'moti';

const from = {
  scale: 0,
  rotate: '0deg',
};

const animate = {
  scale: 1,
  rotate: '45deg',
};

export default function App() {
  return (
      <MotiView
        style={{ backgroundColor: 'green', height: 300, width: 300 }}
        animate={animate}
        from={from}
      />
  );
}
```

I need to investigate why that is, but clearly there is some issue with the animate prop that isn't happening with the `state` prop. I don't exactly understand why. But could you try this solution and confirm that it works for multiple transforms?

Snack comparing the two here: https://snack.expo.dev/@beatgig/moti-rotate-scale-bug

Code:

```tsx
import * as React from 'react';
import { Text, View, Pressable } from 'react-native';
import { MotiView, useAnimationState } from 'moti';

const from = {
  scale: 0,
  rotate: '0deg',
};

const animate = {
  scale: 1,
  rotate: '45deg',
};

export default function App() {
  const state = useAnimationState({
    from,
    to: animate,
  });

  return (
    <Pressable style={{ flex: 1 }}>
      <MotiView
        style={{ backgroundColor: 'blue', height: 300, width: 300 }}
        state={state}
      />
      <MotiView
        style={{ backgroundColor: 'green', height: 300, width: 300 }}
        animate={animate}
        from={from}
      />
    </Pressable>
  );
}
```

**Comment by nandorojo** [comment]
I think I have an idea for what might be causing this. I'll investigate further after #100

**Comment by nandorojo** [comment]
Could you guys please test here and tell me if it's solved: 

```sh
yarn add moti@presence
```

I changed up a bit how I resolve the `from` prop, wondering if that could be it.

**Comment by nandorojo** [comment]
I've found the solution, will push it there soon.

**Comment by nandorojo** [comment]
Fixed in `0.12.0`

```sh
yarn add moti
```
